### PR TITLE
Issue 52 take #2

### PR DESCRIPTION
**Pull request by brunoocasali**
I've created this PR https://github.com/algonauti/ember-g-recaptcha/pull/53 a year ago, after this time a lot of changes were introduced to this library, making that code useless for this new version.

To prevent anyone having their code-breaking because of a push force or something, I created a new branch/PR with the same idea but this time, working with the glimmer version.

In this new PR, I've made everything work as usual and added two use cases without the `@skip` option to make sure the old behavior is set.

Closes https://github.com/algonauti/ember-g-recaptcha/issues/52

CC: @mateusalexandre and @sinankeskin

### CODE REVIEW

**Comment by brunoocasali** [comment]
```
$ ember test
WARNING: Node v17.6.0 is not tested against Ember CLI on your platform. We recommend that you use the most-recent "Active LTS" version of Node.js. See https://git.io/v7S5n for details.
Environment: test
cleaning up...
Built project successfully. Stored in "/var/folders/r2/rh_f917n54v31w6xzbp_35cc0000gp/T/tests-dist-2022211-49302-slmcd9.nc1sq".
(node:49302) [DEP0128] DeprecationWarning: Invalid 'main' field in '/Users/bruno/develop/open-source/ember-g-recaptcha/node_modules/fireworm/package.json' of 'fireworm.js'. Please either fix that or report it to the module author
(Use `node --trace-deprecation ...` to show where the warning was created)
ok 1 Chrome 99.0 - [456 ms] - GRecaptchaComponent: loads real version of recaptcha service
ok 2 Chrome 99.0 - [1658 ms] - GRecaptchaComponent: invokes callbacks successfully
ok 3 Chrome 99.0 - [41 ms] - GRecaptchaComponent > with @skip argument: does not download recaptcha library from Google
ok 4 Chrome 99.0 - [46 ms] - GRecaptchaComponent > with @skip argument: invokes callbacks normally
ok 5 Chrome 99.0 - [0 ms] - ember-qunit: Ember.onerror validation: Ember.onerror is functioning properly

1..5
# tests 5
# pass  5
# skip  0
# todo  0
# fail  0

# ok
```

File: tests/dummy/app/controllers/skip.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export default class SkipController extends Controller {
+  @action
+  auth() {
+    window.grecaptcha.execute();
+  }
+
+  @action
+  forceReset() {
+    this.gRecaptcha.resetReCaptcha();
```
Comment (sinankeskin) [inline tests/dummy/app/controllers/skip.js:12]: We don't have `resetReCaptcha` method it's `reset` now @brunoocasali

Comment (brunoocasali) [reply tests/dummy/app/controllers/skip.js]: cool! changed it :D

**Review by sinankeskin — CHANGES_REQUESTED** [review]
Hey @brunoocasali one small change needs. Could you please add info to changelog for newly added skip parameter and of course please add your name or nickname there. Other than that amazing job here. Thank you for valuable contribution. Appreciated. 🙏

**Comment by brunoocasali** [comment]
@sinankeskin awesome! I did that :D (if you want some change please tell me)

**Comment by sinankeskin** [comment]
Hey @brunoocasali this test fails roughly 4/5 on locally both headless and regular browser. Tested on both Chrome and Chromium based Edge. Could you please take a look? Thanks.

image

**Comment by sinankeskin** [comment]
Also with headless one another test fails time to time like here:

 
image

**Comment by brunoocasali** [comment]
Could you give your ember/node versions?

**Comment by sinankeskin** [comment]
> Could you give your ember/node versions?

image

**Comment by brunoocasali** [comment]
🤔 
image
I'll keep looking to understand what's going on, but I have no idea right now :/

**Comment by brunoocasali** [comment]
Yeah, unfortunately, I didn't make it to reproduce the behavior you showed to me :/ 

```
------ RESULTS ------

Scenario ember-lts-3.24: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.24.3            │ 3.24.6                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-3.28: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.28.0            │ 3.28.8                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-release: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ https://s3.amazon… │ 4.2.0-release+e75bc645       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-beta: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ https://s3.amazon… │ 4.3.0-beta.2.beta+f4b50b51   │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-canary: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ https://s3.amazon… │ 4.4.0-alpha.1.canary+f89dd9… │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-default-with-jquery: SUCCESS
Command run: ember test
with env: {
  "EMBER_OPTIONAL_FEATURES": "{\"jquery-integration\":true}"
}
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ @ember/jquery      │ ^1.1.0             │ 1.1.0                        │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-classic: SUCCESS
Command run: ember test
with env: {
  "EMBER_OPTIONAL_FEATURES": "{\"application-template-wrapper\":true,\"default-async-observers\":false,\"template-only-glimmer-components\":false}"
}
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.28.0            │ 3.28.8                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario embroider-safe: SUCCESS
Command run: ember test
with env: {
  "EMBROIDER_TEST_SETUP_OPTIONS": "safe"
}
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ @embroider/core    │ 1.2.0              │ 1.2.0                        │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ @embroider/webpack │ 1.2.0              │ 1.2.0                        │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ @embroider/compat  │ 1.2.0              │ 1.2.0                        │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ webpack            │ ^5.0.0             │ 5.70.0                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario embroider-optimized: SUCCESS
Command run: ember test
with env: {
  "EMBROIDER_TEST_SETUP_OPTIONS": "optimized"
}
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ @embroider/core    │ 1.2.0              │ 1.2.0                        │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ @embroider/webpack │ 1.2.0              │ 1.2.0                        │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ @embroider/compat  │ 1.2.0              │ 1.2.0                        │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ webpack            │ ^5.0.0             │ 5.70.0                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘


All 9 scenarios succeeded
✨  Done in 609.66s.
```

**Comment by sinankeskin** [comment]
Dropped `node_modules` and tried again after your comment and awkwardly today everything is ok. Thanks again for contribution @brunoocasali 🙏

**Review by sinankeskin — APPROVED** [review]
All good now. Ready to merge 🎉

**Pull request merged**